### PR TITLE
D3ASIM-3512: Fix the SpotMarketTypeEnum to only have two types

### PR DESCRIPTION
**Pull request by Hassan754**

### CODE REVIEW

**Review by spyrostz — APPROVED** [review]
LGTM

**Review by hannesdiedrich — APPROVED** [review]
LGTM. Would be great if you could divide pylint refactorings and actual changes into separate commits for better readability when reviewing. Thanks a lot!

**Review by faizan2590 — DISMISSED** [review]
LGTM

File: d3a_interface/settings_validators.py [hunk 1]
```diff
@@ -37,54 +42,53 @@ def validate_global_settings(settings_dict):
...
         min_slot_length, max_slot_length = calc_min_max_slot_length(tick_length)
         if not min_slot_length <= slot_length <= max_slot_length:
             raise D3ASettingsException("Invalid slot_length "
                                        f"({slot_length.in_minutes()} min, limits: "
                                        f"[{min_slot_length.in_minutes()} min, "
                                        f"{max_slot_length.in_minutes()} min])")
-    if ("cloud_coverage" in settings_dict and
-        not (ConstSettings.PVSettings.CLOUD_COVERAGE_LIMIT.min
-             <= settings_dict["cloud_coverage"] <=
-             ConstSettings.PVSettings.CLOUD_COVERAGE_LIMIT.max)):
```
Comment (faizan2590) [inline d3a_interface/settings_validators.py:57]: Whats the reason to replace `min/max` with indices?

File: d3a_interface/settings_validators.py [hunk 2]
```diff
@@ -37,54 +42,53 @@ def validate_global_settings(settings_dict):
...
-            int(settings_dict["grid_fee_type"]) not in ConstSettings.IAASettings.VALID_FEE_TYPES):
-        raise D3ASettingsException("Invalid value for grid_fee_type "
-                                   f"({settings_dict['grid_fee_type']}).")
-
-    if ("relative_std_from_forecast_percent" in settings_dict and not
-            PercentageRangeLimit.min <=
-            settings_dict["relative_std_from_forecast_percent"] <=
-            PercentageRangeLimit.max):
-        raise D3ASettingsException("Invalid value for relative_std_from_forecast_percent "
-                                   f"({settings_dict['relative_std_from_forecast_percent']}).")
```
Comment (faizan2590) [inline d3a_interface/settings_validators.py:87]: I think these were recent addition in context to settlement market. Why you want to remove it?

**Review by faizan2590 — COMMENTED** [review]
Left two comments to be addressed/answered

File: d3a_interface/settings_validators.py (hunk 2 above)
Comment (Hassan754) [reply d3a_interface/settings_validators.py:87]: It looks like these were messed up in the merge conflict, I'll fix them

File: d3a_interface/settings_validators.py (hunk 1 above)
Comment (hannesdiedrich) [reply d3a_interface/settings_validators.py:57]: @Hassan754 I have the same question. I saw that you reacted with a 👍  but the code did not seem to change.

Comment (hannesdiedrich) [reply d3a_interface/settings_validators.py:57]: weird, github fails me again. I see the commits, but i dont see it in the global diff.
Ignore my comment.

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Review by faizan2590 — APPROVED** [review]
LGTM

**Pull request merged**